### PR TITLE
Editor: Move UndoManager to applicationbase to allow other apps to use it

### PR DESCRIPTION
**Pull request by ResearchDaniel**

### CODE REVIEW

File: src/qt/applicationbase/undomanager.cpp [hunk 1]
```diff
@@ -97,13 +95,6 @@ class AutoSaver {
                 }
 
                 if (str) {
-                    int numRestoreFiles = 100000;
```
Comment (petersteneteg) [inline src/qt/applicationbase/undomanager.cpp:100]: The new version will need to restart inviwo to have this take effect... :( 
Better to pass a lambda if you don't what to couple to the editorsettings

File: src/qt/applicationbase/undomanager.cpp [hunk 2]
```diff
@@ -174,23 +165,24 @@ class AutoSaver {
...
 
 {
 
     QApplication::instance()->installEventFilter(this);
 
-    mainWindow_->getInviwoApplication()->getProcessorNetwork()->addObserver(this);
+    app->getProcessorNetwork()->addObserver(this);
 
-    undoAction_ = new QAction(QIcon(":/svgicons/undo.svg"), QAction::tr("&Undo"), mainWindow_);
+    undoAction_ = new QAction(QIcon(":/svgicons/undo.svg"), QAction::tr("&Undo"), mainWindow);
```
Comment (petersteneteg) [inline src/qt/applicationbase/undomanager.cpp]: It here a reason to parent them to the mainWindow and not just this?

File: src/qt/applicationbase/undomanager.cpp [hunk 3]
```diff
@@ -174,23 +165,24 @@ class AutoSaver {
     std::thread saver_;
 };
 
-UndoManager::UndoManager(InviwoMainWindow* mainWindow)
-    : mainWindow_(mainWindow)
-    , manager_{mainWindow_->getInviwoApplication()->getWorkspaceManager()}
-    , refPath_{filesystem::findBasePath()}
-    , autoSaver_{std::make_unique<AutoSaver>(mainWindow_->getInviwoApplication())}
+UndoManager::UndoManager(InviwoApplication* app, QObject* mainWindow, int numRestoreFiles,
```
Comment (petersteneteg) [inline src/qt/applicationbase/undomanager.cpp]: I would pass just the Network, and the Workspace Manger instead.

**Review by petersteneteg — CHANGES_REQUESTED** [review]

File: src/qt/applicationbase/undomanager.cpp (hunk 1 above)
Comment (ResearchDaniel) [reply src/qt/applicationbase/undomanager.cpp:100]: No change from before

File: src/qt/applicationbase/undomanager.cpp (hunk 2 above)
Comment (ResearchDaniel) [reply src/qt/applicationbase/undomanager.cpp]: Have not tested. 
Could try and see

File: src/qt/applicationbase/undomanager.cpp (hunk 1 above)
Comment (petersteneteg) [reply src/qt/applicationbase/undomanager.cpp:100]: Yes this is a change from before since it was pull on every save before

Comment (ResearchDaniel) [reply src/qt/applicationbase/undomanager.cpp:100]: fixed

File: src/qt/applicationbase/undomanager.cpp (hunk 2 above)
Comment (ResearchDaniel) [reply src/qt/applicationbase/undomanager.cpp]: Tested with this and could not see any difference

File: src/qt/applicationbase/undomanager.cpp (hunk 3 above)
Comment (ResearchDaniel) [reply src/qt/applicationbase/undomanager.cpp]: Done

**Review by petersteneteg — DISMISSED** [review]

**Review by petersteneteg — APPROVED** [review]

**Pull request merged**